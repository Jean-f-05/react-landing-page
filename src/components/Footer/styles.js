import styled, { css } from 'styled-components';
import { Wrapper as FooterElement } from '../../components/TextComponent/styles';
import { Wrapper as SectionContainer } from '../../components/SectionContainer/styles';

export const Wrapper = styled.div`
  ${({ theme }) => css`
  text-align: center;
  border-top: 0.1rem solid ${theme.colors.mediumGray};

    a {
      color: inherit;
      text-decoration:none;
      font-size: ${theme.fonts.small};
      }

  & ${FooterElement} {
    font-size: ${theme.fonts.small};
}

  & ${SectionContainer}{
      padding-top: 0;
      padding-bottom: 0;
  }
`}`;
