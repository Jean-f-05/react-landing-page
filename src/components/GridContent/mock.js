export default {
  title: 'GRID CONTENT',
  html: `<p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Zenonis est, inquam, hoc Stoici. Restinguet citius, si ardentem acceperit. Sed quanta sit alias, nunc tantum possitne esse tanta. Duo Reges: constructio interrete. </p>

  <h2>Similiter sensus, cum accessit ad naturam, tuetur illam quidem, sed etiam se tuetur;</h2>

  <p>Primum in nostrane potestate est, quid meminerimus? Quod iam a me expectare noli. Nihil ad rem! Ne sit sane; Sed haec omittamus; Praeteritis, inquit, gaudeo. </p>`,
  background: false,
};
