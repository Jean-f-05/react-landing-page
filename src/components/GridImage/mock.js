export default {
  background: false,
  title: 'My Grid',
  description:
    'Lorem ipsum dolor sit amet, consectetur adipisicing elit.Ut quasi error, repellat quia, rerum placeat voluptate corrupti veritatis aliquam sapiente animi ipsa illum non vitae similique temporibus quod accusantium quam?',
  grid: [
    {
      altText: 'GRID 1',
      src: 'https://source.unsplash.com/random/300x300',
    },
    {
      altText: 'GRID 2',
      src: 'https://source.unsplash.com/random/300x300',
    },
    {
      altText: 'GRID 3',
      src: 'https://source.unsplash.com/random/300x300',
    },
    {
      altText: 'GRID 4',
      src: 'https://source.unsplash.com/random/300x300',
    },
    {
      altText: 'GRID 5',
      src: 'https://source.unsplash.com/random/300x300',
    },
    {
      altText: 'GRID 6',
      src: 'https://source.unsplash.com/random/300x300',
    },
  ],
};
