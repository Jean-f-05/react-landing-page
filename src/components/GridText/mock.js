export default {
  background: false,
  title: 'My Grid',
  description:
    'Lorem ipsum dolor sit amet, consectetur adipisicing elit.Ut quasi error, repellat quia, rerum placeat voluptate corrupti veritatis aliquam sapiente animi ipsa illum non vitae similique temporibus quod accusantium quam?',
  grid: [
    {
      title: 'GRID 1',
      description:
        'Lorem, ipsum dolor sit amet consectetur adipisicing elit. Temporibus hic neque natus atque ducimus vero fugit. Accusamus sequi molestiae quisquam modi provident cum, ea, nostrum nihil maxime, sunt tempora explicabo.',
    },
    {
      title: 'GRID 2',
      description:
        'Lorem, ipsum dolor sit amet consectetur adipisicing elit. Temporibus hic neque natus atque ducimus vero fugit. Accusamus sequi molestiae quisquam modi provident cum, ea, nostrum nihil maxime, sunt tempora explicabo.',
    },
  ],
};
