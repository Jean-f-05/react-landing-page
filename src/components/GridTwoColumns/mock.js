/* eslint-disable */

export default {
    title: 'GridTwoColumns',
    text: `Lorem ipsum, dolor sit amet consectetur adipisicing elit. Veniam labore totam eaque odio mollitia rerum iusto non ex, obcaecati earum itaque tempore quo autem? Nam quam dolore asperiores tempora illum.`,
    srcImg: "assets/images/javascript.svg",
}