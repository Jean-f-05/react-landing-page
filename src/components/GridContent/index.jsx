import * as S from './styles';
import P from 'prop-types';
import SectionBackground from '../SectionBackground';
import Heading from '../Heading';
import TextComponent from '../TextComponent';

const GridContent = ({ title, html, background = false }) => {
  return (
    <SectionBackground background={background}>
      <S.Wrapper>
        <Heading isUpperCase colorDark={!background}>
          {title}
        </Heading>
        <S.Html>
          <TextComponent>{html}</TextComponent>
        </S.Html>
      </S.Wrapper>
    </SectionBackground>
  );
};

export default GridContent;

GridContent.propTypes = {
  title: P.string.isRequired,
  html: P.string.isRequired,
  background: P.bool,
};
