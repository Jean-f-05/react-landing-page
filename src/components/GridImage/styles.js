import styled, { css } from 'styled-components';
import { Header as HeadingContainer } from '../Heading/styles';
import { Wrapper as TextComponent } from '../TextComponent/styles';

export const Wrapper = styled.div`
  ${({ theme }) => css`
  & > ${TextComponent}{
    margin-bottom: ${theme.spacing.huge};
  }
`}`;

export const Grid = styled.div`
  ${({ theme }) => css`
  display: grid;
  grid-template-columns: repeat(auto-fill,minmax(28rem,1fr));
  gap:${theme.spacing.large}
`}`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    overflow: hidden;
`}`;

export const Image = styled.img`
  ${({ theme }) => css`
    width: 100%;
    transition: all 300ms ease-in-out;

    &:hover {
      transform: scale(1.2) rotate(15deg);
    }
`}`;
