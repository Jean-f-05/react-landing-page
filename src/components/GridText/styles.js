import styled, { css } from 'styled-components';
import { Header as HeadingContainer } from '../Heading/styles';
import { Wrapper as TextComponent } from '../TextComponent/styles';

export const Wrapper = styled.div`
  ${({ theme }) => css`
  ${TextComponent}{
    margin-bottom: ${theme.spacing.huge};
  }
`}`;

export const Grid = styled.div`
  ${({ theme }) => css`
  counter-reset: grid-counter;
  display: grid;
  grid-template-columns: repeat(auto-fill,minmax(28rem,1fr));
  gap:${theme.spacing.large}
`}`;

export const GridElement = styled.div`
  ${({ theme }) => css`
  ${HeadingContainer}{
    position: relative;
    left: 5rem;
  }
  ${HeadingContainer}:before{
    counter-increment: grid-counter;
    content: counter(grid-counter);
    position: absolute;
    font-size: 7rem;
    top: -3rem;
    left: -5rem;
    transform: rotate(5deg);
  }
`}`;
