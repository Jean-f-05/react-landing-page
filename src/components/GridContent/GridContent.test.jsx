import { renderTheme } from '../../styles/renderTheme';
import GridContent from '.';
import { screen } from '@testing-library/react';
import mock from './mock';

describe('<GridContent/>', () => {
  it('should render grid content', () => {
    const { container } = renderTheme(<GridContent {...mock}></GridContent>);
    expect(container).toMatchSnapshot();
  });

  it('should render grid content', () => {
    const { container } = renderTheme(
      <GridContent {...mock} background={undefined}></GridContent>,
    );
    expect(container).toMatchSnapshot();
  });
});
