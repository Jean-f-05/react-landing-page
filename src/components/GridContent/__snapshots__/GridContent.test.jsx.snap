// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridContent/> should render grid content 1`] = `
.c2 {
  text-align: center;
  max-width: 58rem;
  margin: 0 auto;
}

.c4 {
  margin: 6.4rem;
}

.c0 {
  background: #fff;
  color: #0A1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

.c3 {
  color: #0A1128;
  font-size: 6.4rem;
  text-transform: uppercase;
}

.c5 {
  font-size: 2.4rem;
}

@media (max-width:768px) {
  .c3 {
    font-size: 4rem;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h1
          class="c3"
        >
          GRID CONTENT
        </h1>
        <div
          class="c4"
        >
          <div
            class="c5"
          >
            <p>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Zenonis est, inquam, hoc Stoici. Restinguet citius, si ardentem acceperit. Sed quanta sit alias, nunc tantum possitne esse tanta. Duo Reges: constructio interrete. 
            </p>
            

  
            <h2>
              Similiter sensus, cum accessit ad naturam, tuetur illam quidem, sed etiam se tuetur;
            </h2>
            

  
            <p>
              Primum in nostrane potestate est, quid meminerimus? Quod iam a me expectare noli. Nihil ad rem! Ne sit sane; Sed haec omittamus; Praeteritis, inquit, gaudeo. 
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<GridContent/> should render grid content 2`] = `
.c2 {
  text-align: center;
  max-width: 58rem;
  margin: 0 auto;
}

.c4 {
  margin: 6.4rem;
}

.c0 {
  background: #fff;
  color: #0A1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

.c3 {
  color: #0A1128;
  font-size: 6.4rem;
  text-transform: uppercase;
}

.c5 {
  font-size: 2.4rem;
}

@media (max-width:768px) {
  .c3 {
    font-size: 4rem;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h1
          class="c3"
        >
          GRID CONTENT
        </h1>
        <div
          class="c4"
        >
          <div
            class="c5"
          >
            <p>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Zenonis est, inquam, hoc Stoici. Restinguet citius, si ardentem acceperit. Sed quanta sit alias, nunc tantum possitne esse tanta. Duo Reges: constructio interrete. 
            </p>
            

  
            <h2>
              Similiter sensus, cum accessit ad naturam, tuetur illam quidem, sed etiam se tuetur;
            </h2>
            

  
            <p>
              Primum in nostrane potestate est, quid meminerimus? Quod iam a me expectare noli. Nihil ad rem! Ne sit sane; Sed haec omittamus; Praeteritis, inquit, gaudeo. 
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
